sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/MetricExporterConfigurationTest.java

```diff
@@ -49,13 +55,52 @@ void configureReader_PrometheusOnClasspath() {
 
     MetricReader reader =
         MetricExporterConfiguration.configureReader(
-            "prometheus", CONFIG_PROPERTIES, spiHelper, (a, b) -> a, closeables);
+            "prometheus", CONFIG_PROPERTIES, spiHelper, (a, b) -> a, (a, b) -> a, closeables);
     cleanup.addCloseables(closeables);
 
     assertThat(reader).isInstanceOf(PrometheusHttpServer.class);
     assertThat(closeables).hasSize(1);
   }
 
+  @Test
+  void configureReader_customizeReader() {
+    List<Closeable> closeables = new ArrayList<>();
+
+    BiFunction<MetricReader, ConfigProperties, MetricReader> readerCustomizer =
+        (existingReader, config) -> PrometheusHttpServer.builder().setPort(7137).build();
+    MetricReader reader =
+        MetricExporterConfiguration.configureReader(
+            "prometheus", CONFIG_PROPERTIES, spiHelper, readerCustomizer, (a, b) -> a, closeables);
+    cleanup.addCloseables(closeables);
+
+    assertThat(reader).isInstanceOf(PrometheusHttpServer.class);
+    assertThat(closeables).hasSize(2);
+    PrometheusHttpServer prometheusHttpServer = (PrometheusHttpServer) reader;
+    assertThat(prometheusHttpServer)
+        .extracting("httpServer", as(InstanceOfAssertFactories.type(HTTPServer.class)))
+        .satisfies(httpServer -> assertThat(httpServer.getPort()).isEqualTo(7137));
+
+    List<Closeable> closeablesPart2 = new ArrayList<>();
```

Comment:
#nit: Not sure what we gain by having what appears to be two distinct tests (independent setup, assertions) in the same method.